code/modules/plumbing/plumbers/synthesizer.dm

```diff
@@ -61,8 +65,26 @@
 		return
 	if(reagents.total_volume >= amount*delta_time*0.5) //otherwise we get leftovers, and we need this to be precise
 		return
+	if(volume_left < amount) //Empty
+		return
 	reagents.add_reagent(reagent_id, amount*delta_time*0.5)
+	volume_left -= amount
+
+/obj/machinery/plumbing/synthesizer/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/rcd_ammo))
+		var/obj/item/rcd_ammo/R = O
+		if(volume_left > max_volume-100) //no overfilling by more than 100 units
+			to_chat(user, "<span class='warning'>The chemical synthesizer is full!</span>")
+			return ..()
+		volume_left = min(volume_left+R.ammoamt*10, 1000) //400, you get 2x what I originally intended
+		to_chat(user, "<span class='notice'>You refill the chemical synthesizer with the compressed matter cartridge.</span>")
+		qdel(O)
+	else
+		return ..()
```

Comment:
This should work like the RCD in that it only uses as much of the ammo as it needs, rather than consume the entire thing even if only 1 matter is used

Reply:
This is a limitation with RCD ammo, and inserting sheets is just... very weird.

Reply:
I might just make RCD ammo completely refill it instead.

Reply:
Guess I could just edit `R.ammoamt` here, I guess.